azblob/section_writer_test.go

```diff
@@ -0,0 +1,85 @@
+package azblob
+
+import (
+	"bytes"
+	"io"
+
+	chk "gopkg.in/check.v1"
+)
+
+func (s *aztestsSuite) TestSectionWriter(c *chk.C) {
+	b := [10]byte{}
+	buffer := newBytesWriter(b[:])
+
+	section := newSectionWriter(buffer, 0, 5)
+	c.Assert(section.count, chk.Equals, int64(5))
+	c.Assert(section.offset, chk.Equals, int64(0))
+	c.Assert(section.position, chk.Equals, int64(0))
+
+	count, err := section.Write([]byte{1, 2, 3})
+	c.Assert(err, chk.IsNil)
+	c.Assert(count, chk.Equals, 3)
+	c.Assert(section.position, chk.Equals, int64(3))
+	c.Assert(b, chk.Equals, [10]byte{1, 2, 3, 0, 0, 0, 0, 0, 0, 0})
+
+	count, err = section.Write([]byte{4, 5, 6})
+	c.Assert(err, chk.ErrorMatches, "Not enough space for all bytes")
+	c.Assert(count, chk.Equals, 2)
+	c.Assert(section.position, chk.Equals, int64(5))
+	c.Assert(b, chk.Equals, [10]byte{1, 2, 3, 4, 5, 0, 0, 0, 0, 0})
+
+	count, err = section.Write([]byte{6, 7, 8})
+	c.Assert(err, chk.ErrorMatches, "End of section reached")
+	c.Assert(count, chk.Equals, 0)
+	c.Assert(section.position, chk.Equals, int64(5))
+	c.Assert(b, chk.Equals, [10]byte{1, 2, 3, 4, 5, 0, 0, 0, 0, 0})
+
+	section = newSectionWriter(buffer, 5, 6)
```

Comment:
>6 [](start = 39, length = 1)

the 6 is on purpose right? there are only 5 bytes worth of space, but even if we set it to 6, the underlying bytes writer will only write 5 bytes in total. A comment would be helpful here.

Reply:
Good point, thanks! I've added some comments to help clarify this section.